src/granite/migrator.cr

```diff
@@ -48,7 +48,10 @@ module Granite::Migrator
       end
 
       def create
-        resolve = ->(key : String) {
+        resolve = ->(key : String, auto : Bool) {
+          key = key.match(/\((\w+)\,/).not_nil![1] if key.includes?("Union(")
+          key = key.chomp(" | ::Nil") if key.includes?("Nil")
+          key = "AUTO_#{key}" if auto
```

Comment:
Need to come up with a better way to handle this.  I'll think of something.

This also brings the question should the Granite migrator be responsible for setting `NOT NULL` on columns that are not nilable?  Would be added functionality to the migrator, but would require some work, best for another PR later on.